exec/java-exec/src/test/java/org/apache/drill/exec/compile/TestLargeFileCompilation.java

```diff
@@ -154,4 +158,20 @@ public void testProject() throws Exception {
     testNoResult(ITERATION_COUNT, LARGE_QUERY_SELECT_LIST);
   }
 
+  @Test
+  public void testSelectAllFromFileWithManyColumns() throws Exception {
+    File path = new File(BaseTestQuery.getTempDir("json/input"));
```

Comment:
You do not have to generate a new json file for writing a unit test. Not sure how the new created file is cleaned up if the testcase failed.

You may consider using the existing tpch sample file to do unit test, or re-enable the originally ignored test (testTOP_N_SORT).

```java
  @Test
  public void testLargeListColWithLimit() throws Exception {
    final int nCol = 1000;
    final StringBuilder sb = new StringBuilder();

    sb.append(" select n_nationkey ");
    for (int i = 0; i< nCol; i++) {
      sb.append(", " + "Col" + i );
    }
    sb.append(" from cp.`tpch/nation.parquet`");
    sb.append(" limit 1");

    test(sb.toString());
  }
```